src/client-scopes/details/MappingDetails.tsx

```diff
@@ -163,199 +169,76 @@ export const MappingDetails = () => {
-                          key={option}
-                          value={option}
-                        />
-                      ))}
-                </Select>
-              )}
-            />
-          </FormGroup>
-          <FormGroup
-            hasNoPaddingTop
-            label={t("addClaimTo")}
-            fieldId="addClaimTo"
-          >
-            <Flex>
-              <FlexItem>
-                <Controller
-                  name="config.id-token-claim"
-                  defaultValue="false"
-                  control={control}
-                  render={({ onChange, value }) => (
-                    <Checkbox
-                      label={t("idToken")}
-                      id="idToken"
-                      isChecked={value === "true"}
-                      onChange={(value) => onChange("" + value)}
-                    />
-                  )}
-                />
-              </FlexItem>
-              <FlexItem>
-                <Controller
-                  name="config.access-token-claim"
-                  defaultValue="false"
-                  control={control}
-                  render={({ onChange, value }) => (
-                    <Checkbox
-                      label={t("accessToken")}
-                      id="accessToken"
-                      isChecked={value === "true"}
-                      onChange={(value) => onChange("" + value)}
-                    />
-                  )}
+                helperTextInvalid={t("common:required")}
+              >
+                <TextInput
+                  ref={register({ required: true })}
+                  type="text"
+                  id="name"
+                  name="name"
+                  validated={
+                    errors.name
+                      ? ValidatedOptions.error
+                      : ValidatedOptions.default
+                  }
                 />
-              </FlexItem>
-              <FlexItem>
-                <Controller
-                  name="config.userinfo-token-claim"
-                  defaultValue="false"
-                  control={control}
-                  render={({ onChange, value }) => (
-                    <Checkbox
-                      label={t("userInfo")}
-                      id="userInfo"
-                      isChecked={value === "true"}
-                      onChange={(value) => onChange("" + value)}
-                    />
-                  )}
-                />
-              </FlexItem>
-            </Flex>
-          </FormGroup>
+              </FormGroup>
+            </>
+          )}
+          <FormProvider {...form}>
+            {mapping?.properties.map((property) => {
+              const componentType = property.type!;
+              if (ComponentTypes.includes(componentType)) {
+                const Component = COMPONENTS[componentType as Components];
+                return <Component key={property.name} {...property} />;
+              } else {
```

Comment:
Instead of using `includes()` with a type-cast you could write a function that uses a [type-guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to narrow down the type. For example:

```ts
const isValidComponentType = (value: string): value is Components => value in COMPONENTS;
```

Now you can use this function to narrow down the type without needing to cast it.
```suggestion
              if (isValidComponentType(componentType)) {
                const Component = COMPONENTS[componentType];
                return <Component key={property.name} {...property} />;
              } else {
```

Reply:
Yes it's a cast, but it can never fail as Components are all the Components that are in ComponentTypes. And your function is also casting by specifying the return type.